controllers/etcdpeer_controller.go

```diff
@@ -292,6 +291,8 @@ func (r *EtcdPeerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&etcdv1alpha1.EtcdPeer{}).
 		// Watch for changes to ReplicaSet resources that an EtcdPeer owns.
 		Owns(&appsv1.ReplicaSet{}).
-		Owns(&corev1.PersistentVolumeClaim{}).
+		// We can use a simple EnqueueRequestForObject handler here as the PVC
+		// has the same name as the EtcdPeer resource that needs to be enqueued
+		Watches(&source.Kind{Type: &corev1.PersistentVolumeClaim{}}, &handler.EnqueueRequestForObject{}).
```

Comment:
This causes a reconciliation for PVCs that are not owned by use. Copied from https://github.com/munnerz/etcd-operator-poc/blob/master/controllers/etcdpeer_controller.go#L251

But I think it's going to trigger Reconcile with the names of every PVC in the cluster, which won't do any harm except that it wakes up the operator more often than necessary and it'll result in loads of not-found errors in the logs.
So I probably need to refine it to check the name matches the `For` peer.

Reply:
I've added a custom event handler here.